azure-pipelines.yml

```diff
@@ -29,15 +29,15 @@ jobs:
     name: Linux
     vmImage: ubuntu-18.04
     matrix:
-      py37_np118_32bit:
+      py38_np118_32bit:
         # 32 bit linux only has np 1.15
-        PYTHON: '3.7'
+        PYTHON: '3.8'
         NUMPY: '1.18'
         CONDA_ENV: azure_ci
         BITS32: yes
         TEST_START_INDEX: 2
```

Comment:
@esc Would you like me to go ahead and proceed with removing this 32-bit job in the matrix?

re [failure](https://dev.azure.com/numba/numba/_build/results?buildId=12756&view=logs&jobId=5f38b89c-308c-5e22-4982-c0a0f2a8cf5c&j=5f38b89c-308c-5e22-4982-c0a0f2a8cf5c&t=29dd518e-3410-5392-c104-01ec46cfe18b)

Reply:
I am not sure yet, I am hoping to discuss this during the developer meeting today. Hopefully, we can find a consensus soon.

Reply:
Why was the job called `py37_np118_32bit` if the comment says `32 bit linux only has np 1.15`? (How could it have been working / passing tests, given that there will be differences in implementation between Numba and NumPy - was it just getting lucky and not hitting tests in which there would have been a difference?)

Reply:
I think that comment is from a time when `linux-32` conda packages were available, NumPy 1.15 being the last version. NumPy 1.18 comes as a wheel:
https://github.com/numba/numba/blob/fc28b2974904c0e1e02b62a9cf37f75023cfc80d/buildscripts/incremental/setup_conda_environment.sh#L84-L87

Reply:
Thanks for the clarification. I think the comment might as well be removed - it could be updated to say that it only has/had 1.15 conda packages available, but I'm not sure it's useful information to have here.

Reply:
Ok yes I was a bit confused by that code comment as well.

I see why the 32bit CI job is failing now. There are no 32bit conda packages for supporting pip/gitpython/pyaml for Python 3.8, not just NumPy. They would all have to come as whl files from PyPI by pip installing them.

Should this conda env job still be run if there aren’t going to be 32bit Linux conda packages to use?

Reply:
> I see why the 32bit CI job is failing now. There are no 32bit conda packages for supporting pip/gitpython/pyaml for Python 3.8, not just NumPy. They would all have to come as whl files from PyPI by pip installing them.

This is correct. In fact, IIRC, the last Python version to be supported for linux-32 via the Anaconda Distro Team is Python 3.7 -- so we don't even have a way to obtain a Python for this platform.

> Should this conda env job still be run if there aren’t going to be 32bit Linux conda packages to use?

Probably not, no. As mentioned previously: I hope to get consensus on this during todays developer meeting. As soon as we agree, we can give you instructions on how to proceed.

Reply:
@jamesobutler we decided to remove support for the following come 0.57 (current estimated release date: Jan 2023):

* Python below 3.8
* Numpy below 1.19
* All 32 bit architectures

However, since the stencil tests depend on the 3.7 AST, we need to get the refactor #8325 in, before we can remove 3.7 testing from the public CI matrix. You can go ahead and remove the 32 bit test for now, but the merge of this PR will delayed a little.

Reply:
@esc I saw that you approved #8325 and that it is ready to merge. I have gone ahead and pushed another commit to this PR that removes the 32bit testing in the CI. All checks have now passed 🎉. Let me know if you would like additional changes here in this PR. Here to help!

Reply:
@jamesobutler yes, thank you very much. #8325 has now been merged and I hope to look at this one and #8318 and #7920 early next week. 🙏